src/RL/utils/geometricflux/graphconv.jl

```diff
@@ -1,23 +1,28 @@
 
-struct GraphConv{A <: AbstractMatrix, B}
+
+struct GraphConv{A <: AbstractMatrix, B, G <: pool}
     weight1::A
     weight2::A
     bias::B
     σ
+    pool::G
 end
 
-function GraphConv(ch::Pair{Int, Int}, σ=identity; init=Flux.glorot_uniform, bias::Bool=true, T::DataType=Float32)
+function GraphConv(ch::Pair{Int, Int}, σ=Flux.leakyrelu; init=Flux.glorot_uniform, bias::Bool=true, T::DataType=Float32, pool::pool=meanPooling())
     in, out = ch
     W1 = init(out, in)
     W2 = init(out, in)
     b = bias ? T.(init(ch[2])) : zeros(T, ch[2])
-    return GraphConv(W1, W2, b, σ)
+    return GraphConv(W1, W2, b, σ, pool)
 end
 
 Flux.@functor GraphConv
 
 function (g::GraphConv)(fgs::BatchedFeaturedGraph{Float32}) 
     A, X = fgs.graph, fgs.nf
+    if isa(g.pool,meanPooling)
```

Comment:
The convolutions should be compatible with sum pooling. Eventually, passing through parametric types with different functions would be a solution.

That is, write two functions that are more or less typed like this:
- `function (g::GraphConv{A, B, meanPooling})(fgs::BatchedFeaturedGraph{Float32})`
- `function (g::GraphConv{A, B, sumPooling})(fgs::BatchedFeaturedGraph{Float32})`

This would avoid ifs in the function, which I think is not easy to handle with Zygote.